### ISSUE TITLE
Package Request: ros-noetic-desktop-full / ros-noetic-desktop

### ISSUE THREAD

**Issue by amralmorsi** [arch4edu/arch4edu#176]
Quoting from the ROS package's page on ArchWiki: "ROS is an open-source, meta-operating system for your robot. It provides the services you would expect from an operating system, including hardware abstraction, low-level device control, implementation of commonly-used functionality, message-passing between processes, and package management."

ROS's page on ArchWiki: [https://wiki.archlinux.org/title/ROS](https://wiki.archlinux.org/title/ROS)
ROS's page in AUR: [https://aur.archlinux.org/packages/ros-noetic-desktop-full/](https://aur.archlinux.org/packages/ros-noetic-desktop-full/)

Honestly, I find it odd this package doesn't exist on arch4edu yet. I remember I have read somewhere it exists in arch4edu, but I can't find it for some reason. (And actually the package's [official website](https://ros.org) excplicitly says the pervious version exists on arch4edu, but I can't find this one either).

**Comment by petronny** [comment]
Oh the ros 1 noetic has been replaced by ros 2 humble #149 .
And the old noetic packages have been removed yesterday.
 
I have a concern about adding it back, which is many noetic packages are orphaned on AUR.
I'm not sure if some of these unmaintained packages will break the build of ros-noetic-desktop-full. @acxz Could you tell us more about the states of the noetic packages?

**Comment by petronny** [comment]
Just find that none of the noetic packages is orphaned now. I'll try add them back.

**Comment by acxz** [comment]
@petronny morning here from the US. Hope things are going well with you!

ROS 1 and ROS 2 are very different and don't operate with each other without the help of external packages.

Melodic (older ROS 1) and Noetic (latest ROS 1) share very much the same code and as such melodic has been orphaned and for Arch users that want to use ROS 1, we point them to the latest ROS 1 Noetic. Noetic EOL's 2025 and I plan to maintain the packages till then.

ROS 2 on the other hand can be thought of as completely separate and the introduction of a ROS 2 version does not mean we can orphan a ROS 1 version, like we could for ROS 1 versions between themselves.

At some point or another I plan to package ROS 2 rolling as that ROS 2 version better fits Arch's philosophy (https://github.com/ros-rolling-arch). I will most likely start doing that when my current research project involving noetic is finished and I can port the stack to rolling.

Hopefully that clear's things about between ROS 1 and ROS 2.

tldr; noetic will be maintained till 2025, as such I would love if arch4edu can keep providing binaries till then

**Comment by petronny** [comment]
[c5d6d8d](https://github.com/arch4edu/arch4edu/commit/c5d6d8dfc4a1e40f4021a8ae5e9c89ca06fc7c09)
